src/python/dxpy/dx_extract_utils/cohort_filter_payload.py

```diff
@@ -22,18 +22,30 @@ def generate_pheno_filter(values, entity, field, filters):
         raise ValueError("Invalid input cohort. Cohorts must have “and” logic on the primary entity and field.")
 
     entity_field = "$".join([entity, field])
-    entity_field_filter = {"condition": "in", "values": values}
 
     # Search for an existing filter for the entity and field
     for compound_filter in filters["pheno_filters"]["compound"]:
         if "filters" not in compound_filter or entity_field not in compound_filter["filters"]:
             continue
         if "logic" in compound_filter and compound_filter["logic"] != "and":
             raise ValueError("Invalid input cohort. Cohorts must have “and” logic on the primary entity and field.")
-        # The entity field filter is valid for addition of the "in" values condition
-        compound_filter["filters"][entity_field].append(entity_field_filter)
+        primary_filters = []
+        for primary_filter in compound_filter["filters"][entity_field]:
+            if "condition" not in primary_filter or primary_filter["condition"] == "exists":
+                pass
+            elif primary_filter["condition"] == "in":
+                values = [value for value in values if value in primary_filter["values"]]
```

Comment:
Will using set.intersect be faster here? Or any other method, since I can imagine the number of IDs can be very large sometimes. Similarly below is it better to use set(A)-set(B)?

Reply:
I was trying to preserve order here so I didn't use set intersection/subtract here. Since there can be a large number of IDs I will update this.